lib/matplotlib/legend.py

```diff
@@ -535,6 +539,17 @@ def val_or_rc(val, rc_name):
                     loc = locs[0] + ' ' + locs[1]
             # check that loc is in acceptable strings
             loc = _api.check_getitem(self.codes, loc=loc)
+        elif isinstance(loc, tuple):
```

Comment:
I'm wondering if we should be more lenient here: currently there is nothing to stop a user passing e.g. a 2-element list, and it works just fine, but this will presumably break anyone doing so.  Not long ago we had a similar discussion about the iterable passed to `colorbar`, and ultimately decided to be more lenient than had previously been documented https://github.com/matplotlib/matplotlib/pull/24408.

Reply:
This seems reasonable to me. If we allow a list we should:

1. check the length and error on the wrong length. 

2. update the docs and make it a supported type. 

What do you think?

Reply:
For maximum flexibility, we could attempt to convert anything that isn't `str` or `int` into a tuple within a try-except loop.  Then just use your existing handling for tuple.  Probably better to wait and see what other code developers think though.

Reply:
I went ahead an added the case for lists and then convert the list into a tuple. The tuple logic then evaluates the length, etc. 

I changed my mind on the documentation. It seems like we are only allowing lists for backward compatibility. Tuples are probably the better choice and so the api should probably advertise tuples as the valid type for coordinate pairs.